js/modules/k6/html/html.go

```diff
@@ -71,3 +99,438 @@ func (s Selection) Attr(name string, def ...goja.Value) goja.Value {
+
+		default:
+			panic(s.rt.NewGoError(errors.New("Invalid argument. The selector must be a string or query object")))
+			return Selection{}
+		}
+	} else {
+		filter := def[1].String()
+		switch selector.(type) {
+		case string:
+			return Selection{s.rt, filteredUntil(filter, selector.(string))}
+
+		case Selection:
+			return Selection{s.rt, filteredUntilSelection(filter, selector.(Selection).sel)}
+
+		case nil:
+			return Selection{s.rt, filteredUntil(filter, "")}
+
+		default:
+			panic(s.rt.NewGoError(errors.New("Invalid argument. The selector must be a string or query object")))
+			return Selection{}
+		}
+	}
+}
+
+// prevUntil, nextUntil and parentsUntil support two args
+// 1st arg is either a selector string, goquery selection object, or nil
+// 2nd arg is filter selector string or nil/undefined
+func (s Selection) PrevUntil(def ...goja.Value) Selection {
+	return s.adjacentUntil(
+		s.sel.PrevUntil,
+		s.sel.PrevUntilSelection,
+		s.sel.PrevFilteredUntil,
+		s.sel.PrevFilteredUntilSelection,
+		def...,
+	)
+}
+
+func (s Selection) NextUntil(def ...goja.Value) Selection {
+	return s.adjacentUntil(
+		s.sel.NextUntil,
+		s.sel.NextUntilSelection,
+		s.sel.NextFilteredUntil,
+		s.sel.NextFilteredUntilSelection,
+		def...,
+	)
+}
+
+func (s Selection) ParentsUntil(def ...goja.Value) Selection {
+	return s.adjacentUntil(
+		s.sel.ParentsUntil,
+		s.sel.ParentsUntilSelection,
+		s.sel.ParentsFilteredUntil,
+		s.sel.ParentsFilteredUntilSelection,
+		def...,
+	)
+}
+
+func (s Selection) Slice(start int, def ...int) Selection {
+	if len(def) > 0 {
+		return Selection{s.rt, s.sel.Slice(start, def[0])}
+	} else {
+		return Selection{s.rt, s.sel.Slice(start, s.sel.Length())}
+	}
+}
+
+func (s Selection) Get(def ...int) goja.Value {
+	if len(def) == 0 {
+		var items []goja.Value
+
+		for i := 0; i < len(s.sel.Nodes); i++ {
+			items = append(items, selToElement(s.Eq(i)))
+		}
+
+		return s.rt.ToValue(items)
+	} else if def[0] < s.sel.Length() && def[0] > -s.sel.Length() {
+		return selToElement(s.Eq(def[0]))
+	} else {
+		return goja.Undefined()
+	}
+}
+
+func (s Selection) ToArray() (items []Selection) {
+	for i := range s.sel.Nodes {
+		items = append(items, Selection{s.rt, s.sel.Eq(i)})
+	}
+	return
+}
+
+func (s Selection) Size() int {
+	return s.sel.Length()
+}
+
+func (s Selection) Index(def ...goja.Value) int {
+	if len(def) == 0 {
+		return s.sel.Index()
+	}
+
+	v := def[0].Export()
+	switch v.(type) {
+	case Selection:
+		return s.sel.IndexOfSelection(v.(Selection).sel)
+
+	case string:
+		return s.sel.IndexSelector(v.(string))
+
+	case Element:
+		if elem, ok := def[0].Export().(Element); ok {
+			return s.sel.IndexOfNode(elem.node)
+		} else {
+			return -1
+		}
+	default:
+		return -1
+	}
+}
+
+// end result of the following is two strings.Replacer objects
+// Replacer("-a", "A", "-b", "B"..., "-z", "Z") and Replacer("A", "-a",...)
+//to translate to "data-attr-name" to "attrName" and back
+const (
```

Comment:
Put all of this at the top of the file instead, constants first, then helpers.

But before doing that: can you refactor this to use `strings` functions instead? That'll handle unicode properly as well.